File: comms/push-notifications/device-registrations/list-device-registrations/list-device-registrations.go
package example

import (
    context "context"

    twiliocomms "github.com/twilio/twilio-comms-go/twilio"
    client "github.com/twilio/twilio-comms-go/twilio/client"
    option "github.com/twilio/twilio-comms-go/twilio/option"
)

func do() {
    client := client.NewWithOptions(
        option.WithBasicAuth(
            "TWILIO_ACCOUNT_SID",
            "TWILIO_AUTH_TOKEN",
        ),
    )
    request := &twiliocomms.DeviceRegistrationsListRequest{
        UserId: "comms_pushnotificationuser_01h9krwprkeee8fzqspvwy6nq8",
        AppName: twiliocomms.String(
            "appName",
        ),
        PageToken: twiliocomms.String(
            "pageToken",
        ),
        PageSize: twiliocomms.Int(
            50,
        ),
    }
    client.PushNotifications.DeviceRegistrations.List(
        context.TODO(),
        request,
    )
}

File: comms/push-notifications/device-registrations/list-device-registrations/list-device-registrations.cs
using Twilio.Comms;
using System.Threading.Tasks;
using Twilio.Comms.PushNotifications;

public partial class Examples
{
    public async Task Example() {
        var client = new TwilioComms(
            accountId: "TWILIO_ACCOUNT_SID",
            authToken: "TWILIO_AUTH_TOKEN"
        );

        await client.PushNotifications.DeviceRegistrations.ListAsync(
            new DeviceRegistrationsListRequest {
                UserId = "comms_pushnotificationuser_01h9krwprkeee8fzqspvwy6nq8",
                AppName = "appName",
                PageToken = "pageToken",
                PageSize = 50
            }
        );
    }

}
